homeassistant/components/vicare/config_flow.py

```diff
@@ -60,7 +72,62 @@ async def async_step_user(
 
         return self.async_show_form(
             step_id="user",
-            data_schema=vol.Schema(data_schema),
+            data_schema=USER_SCHEMA,
+            errors=errors,
+        )
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle re-authentication with ViCare."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm re-authentication with ViCare."""
+        errors: dict[str, str] = {}
+        assert self.entry is not None
+
+        if user_input is not None:
+            password = user_input[CONF_PASSWORD]
+            client_id = user_input[CONF_CLIENT_ID]
+            data = {
+                CONF_USERNAME: self.entry.data[CONF_USERNAME],
+                CONF_PASSWORD: password,
+                CONF_CLIENT_ID: client_id,
+                CONF_HEATING_TYPE: self.entry.data[CONF_HEATING_TYPE],
+            }
+
+            try:
+                await self.hass.async_add_executor_job(vicare_login, self.hass, data)
+            except PyViCareInvalidCredentialsError:
+                errors["base"] = "invalid_auth"
+            else:
+                self.hass.config_entries.async_update_entry(
+                    self.entry,
+                    data={
+                        **self.entry.data,
+                        CONF_PASSWORD: password,
+                        CONF_CLIENT_ID: client_id,
+                    },
+                )
+                await self.hass.config_entries.async_reload(self.entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+        schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_PASSWORD, default=self.entry.data[CONF_PASSWORD]
+                ): cv.string,
+                vol.Required(
+                    CONF_CLIENT_ID, default=self.entry.data[CONF_CLIENT_ID]
+                ): cv.string,
```

Comment:
Are defaults allowed here as this is giving the password in plain text? At least for the client ID it is convenient for me to have that printed out to double check.

Reply:
It's up to the integration to show the password or not

Reply:
Ok. How could this work with your suggested changes? I need to set the default just for this usage.

Reply:
Ah, missed one suggestion that covers this, I'll try that.